src/Features/CSharp/Portable/CodeFixes/Nullable/CSharpDeclareAsNullableCodeFixProvider.cs

```diff
@@ -190,12 +205,23 @@ static TypeSyntax TryGetSingleTypeArgument(TypeSyntax type)
             }
         }
 
+        private static bool IsExpressionSupported(SyntaxNode node)
+        {
+            return node.IsKind(
```

Comment:
@ryzngard @KristianJakubik I recognize I'm reviewing this late but didn't see this one go by until I saw some bug statuses get updated -- why is this doing syntactic checking here versus just checking if the expression is nullable according to flow state? If I have a function that's doing:

```
string GetSomething()
{
    string? s = SomethingElseThatReturnsNullable();
    return s;
}
```

Don't I want that to trigger as well?

Reply:
@jasonmalinowski can you file a separate issue to follow up on this? I don't believe this should be backed out, but the scenario is overlooked.

Reply:
@jasonmalinowski 
Yeah you are right, it would make much more sense to handle it from flow state point. Just all the issues filled in were focused on particular syntactic cases, so it felt natural to me, handle them separately. From big picture doesn't make sense so much anymore.
Anyway thanks for feedback.

Reply:
Yeah, the issues somewhat primed a whack-a-mole approach hence my concern. @sharwell did have a suggestion though that we might want to have different fix all behaviors though. The classes of issues you fixed are ones that are very "obvious" that the shape of the API needs changing, i.e. if you see "return <condition> ? <something> : null" it's plain as day the API needs to be marked as a nullable return, and so that might be useful to have a fix all for all of that at once. For my case, it's not quite as obvious -- maybe the code in the function needs updating, maybe not. The "confidence" is absolutely lower and there might need to be some accounting for that.

I'll file a bug in either case to track but I absolutely think that [a] the work was great and so thank you for that and [b] we might still have the special conditional code in there, it just gets moved around a bit.